proposals/0002-gep.md

```diff
@@ -0,0 +1,140 @@
+---
+MEP: 0002
+Title: Unifed API
+Discussion: Link
+Implementation: Link
+---
+
+# TITLE: Unified API
+
+## Abstract
+
+This GEP discusses the unified API that handles the interaction with LLM providers. This represents the feature set for MVP.
+
+## Motivation
+
+
+The Unified API abstracts much of the overhead of interacting with LLM providers.
+It stores all the provider specific logic in a single place and allowing developers to focus on the core logic. It also introduces durability to LLM apps. Additionally, unifying the API allows optimizations to be leveraged such as latency and cost improvements.
+
+
+### Requirements
+
+- R1: User should be able to specify specific models or select auto mode (selection handled by Glide)
+- R2: Should be a Hertz Client
+- R3: Should take a user request as input
+- R4: Handles all provider specific logic
+- R5: Easily maintained
+- R6: API schemas must unify common LLM request params (e.g. topP, N, etc)
+- R7: API routes must unify common LLM endpoints/API
+- R8: API schemas should allow to customize provider-specific params (e.g. a different messages for OpenAI/Anthropic)
+
+
+## Design
+
+
+- A Hertz client that handles an incoming request and returns a reponse from that provider.
+- Provider specific clients
+
+### Routes
+```yaml
+routes:
+    chat: /chat/{pool-id}
+    complete: /complete/{pool-id} #this is also know as generate in some providers
+    multi-modal: /multimodal/{pool-id} # future route, not for MVP
+    embedding: /embedding/{pool-id} # future route, not for MVP
+```
+
+#### User Request Schema for Chat Route
+
+Below is an example user request to the Chat route. The other routes will look very similar, if not identical. The provider specific params will look different.
+
+Example Request Body:
+
+
+```json
+{
+  "message": "tell me a joke",
+}
+```
+
+### Provider Request Schemas
```

Comment:
Could you please add one more request/response schemas for another provider to be able to see common ground?